### PR TITLE
Refresh credentials on 50 minutes max

### PR DESCRIPTION
**Pull request by elorzafe**
_Issue #, if available:_
#4188

_Description of changes:_
Forcing Cognito credentials to refreshed on 50 minutes max since it was retrieved from the service.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Christilut** [comment]
Can this be merged asap? Currently my application is logging users out at seemingly random times and it's even happening once or twice every day during development.

File: packages/core/__tests__/Credentials-test.ts [hunk 1]
```diff
@@ -70,10 +64,12 @@ describe('Credentials test', () => {
 			Amplify.register(cacheClass);
 			const credentials = new Credentials(null);
 
+			const expireTime = new Date().getTime() + 20 * 60 * 1000;
```
Comment (ericclemmons) [inline packages/core/__tests__/Credentials-test.ts]: @elorzafe  Should this line be updated to use `DateUtils.getDateWithOffset().getTime()` from #4844?

Comment (elorzafe) [reply packages/core/__tests__/Credentials-test.ts]: I think is not necessary because the time is only local, not sent or compared to something external

Comment (ericclemmons) [reply packages/core/__tests__/Credentials-test.ts]: Excellent point, since this is just for comparison down below.

**Pull request merged**